pkgs/misc/screensavers/betterlockscreen/default.nix

```diff
@@ -0,0 +1,46 @@
+{
+  stdenv, makeWrapper, fetchFromGitHub, substituteAll,
+  imagemagick, i3lock-color, xdpyinfo, xrandr, bc, feh
+}:
+
+stdenv.mkDerivation rec {
+  name = "betterlockscreen-${version}";
+  version = "3.0.1";
+
+  src = fetchFromGitHub {
+    owner = "pavanjadhaw";
+    repo = "betterlockscreen";
+    rev = version;
+    sha256 = "0jc8ifb69shmd0avx6vny4m1w5dfxkkf5vnm7qcrmc8yflb0s3z6";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  patches = [ (substituteAll {
+    src = ./replace-i3lock.patch;
+  })];
+
+  # unpack fetches and unpatcks
+  # patch patches for i3lock to i3lock-color
+  # install wraps the program, with all the dependencies
+  phases = ["unpackPhase" "patchPhase" "postInstall"];
+
+  postInstall = 
```

Comment:
Rename `postInstall` to `installPhase` (https://nixos.org/nixpkgs/manual/#ssec-install-phase) if `make install` doesn't otherwise do something.